### ISSUE TITLE
Deprecate JavaParserFacade

### ISSUE THREAD

**Issue by matozoid** [javaparser/javaparser#1377]
That stuff can be done through ParserConfiguration now.

**Comment by malteskoruppa** [comment]
In this case I would suggest to also add some documentation giving an example on how to resolve types now.

The JavaSymbolSolver README still suggests to proceed as follows (https://github.com/javaparser/javasymbolsolver#how-can-i-use-it-show-me-some-code):

```
Node node = <get this node by parsing source code with JavaParser>
Type typeOfTheNode = JavaParserFacade.get(typeSolver).getType(node);
```
That's exactly what I've been doing up to now. If JavaParserFacade gets deprecated, it's fine by me, but then how should I proceed to resolve the type of a node now? :smiley:

**Comment by matozoid** [comment]
Yes, even I didn't know - the integration went quickly, but the documentation hasn't been updated anywhere yet. All I have is https://github.com/javaparser/javasymbolsolver-maven-sample

**Comment by malteskoruppa** [comment]
Ah, I see... thanks for the link. So now we can just call `calculateResolvedType()` on a node after setting up the symbol solver in the JavaParser configuration, and we don't have to pass around the typeSolver all the time. Very nice! I'll give it a try as soon as I get around to it. :smile:

**Comment by matozoid** [comment]
I'm assuming the new integrated way has the same functionality as the previous way - @ftomassetti ?

**Comment by dhardtke** [comment]
Suppose I have a `TypeSolver` that is depending on some external state (e.g., I pass the type solver some source code of classes as Strings instead of reading them from a file): Am I right assuming that I can no longer pass around a `JavaParserFacade`, but instead need to pass around a `JavaParser` instance that is configured with the appropriate `TypeSolver` / `JavaSymbolSolver`?

I think it might instead be a good idea to manually pass my methods a `JavaSymbolSolver` instance and the previously already passed `CompilationUnit` and just stick to the Javadoc here: https://github.com/javaparser/javaparser/blob/master/javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/JavaSymbolSolver.java#L26

(Although I am not sure how likely it is that API is going to change).

EDIT:
Another question: Is it safe to assume the `CombinedTypeSolver` is **not** thread-safe because it is forming a hierarchy of `TypeSolvers` using the `parent` attribute and, hence, I can not share the same `TypeSolver` instances (**not** `CombinedTypeSolver`s) between multiple `CombinedTypeSolver`s?

**Comment by ftomassetti** [comment]
The new integration does one thing: hide the typesolver in the AST. Then there are additional methods on the nodes (like `calculateResolvedType`) that use that typeSolver and call JavaParserFacade behind the scene. In this sense it is discourages to use JavaParserFacade directly, because the new API (basically new methods in the AST nodes) should be always nicer and more intuitive to use.

**Comment by matozoid** [comment]
Lots of questions there... The only one I can say something about is that you could pass a `ParserConfiguraiton` around instead of `JavaParser` itself.
